source/core/shape_parsing.cc

```diff
@@ -154,7 +154,29 @@ ShapeGenerator::initialize_shape_from_file(const std::string   type,
           // All the information is concatenated into only one object so
           // that the usual initialization function can be called.
           std::map<std::string, std::vector<double>> rbf_data;
-          fill_vectors_from_file(rbf_data, file_name, " ");
+
+          // This step is used to parse an optional argument in the
+          // file_name. If the shape argument is <filename>;<number of levels
+          // not precalculated>, then it is parsed. Otherwise, if the shape
+          // argument is only <filename>, then the default value of number of
+          // levels not precalculated (=0) is assumed.
+          std::vector<std::string> list_of_words_base =
+            Utilities::split_string_list(file_name, ";");
+          std::vector<std::string> list_of_words_clean;
+          for (unsigned int j = 0; j < list_of_words_base.size(); ++j)
+            {
+              if (list_of_words_base[j] != "")
+                {
+                  list_of_words_clean.push_back(list_of_words_base[j]);
+                }
+            }
+          std::string true_file_name           = list_of_words_clean[0];
+          double      levels_not_precalculated = 0.;
+          if (list_of_words_clean.size() > 1)
+            levels_not_precalculated =
+              Utilities::string_to_double(list_of_words_clean[1]);
+
```

Comment:
This parser is getting messier and messier. I must admit I am not a fan of this.
Consider moving this to a more standard format, prm or json or something.
Maybe not now, but it will need to happen eventually. The longer you wait, the more painful it will be.

Reply:
I agree with you, I'll look into it for a future short PR